test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/Log4NetInputTests.cs

```diff
@@ -0,0 +1,101 @@
+                    {
+                        ["logLevel"] = "Verbose"
+                    }
+                }
+            };
+
+            using (var configFile = new TemporaryFile())
+            {
+                var pipelineConfig = JsonConvert.SerializeObject(pipelineConfigObj);
+
+                configFile.Write(pipelineConfig);
+                var configBuilder = new ConfigurationBuilder();
+                configBuilder.AddJsonFile(configFile.FilePath);
+                var configuration = configBuilder.Build();
+                var inputsConfigSection = configuration.GetSection("inputs");
+                var configFragments = inputsConfigSection.GetChildren().ToList();
+
+                var l4nInputConfiguration = new Log4netConfiguration();
+                configFragments[0].Bind(l4nInputConfiguration);
+
+                Assert.Equal("Verbose", l4nInputConfiguration.LogLevel);
+            }
+        }
+
+        [Fact]
+        public void ValidConfigurationCanWrtieToInputPipleine()
+        {
+            var healthReportMock = new Mock<IHealthReporter>();
+            var configurationMock = new Log4netConfiguration { LogLevel = "INFO" };
+            var mockOutput = new Mock<IOutput>();
+
+            using (var log4NetInput = new Log4netInput(configurationMock, healthReportMock.Object))
+            using (var pipeline = new DiagnosticPipeline(
+                healthReportMock.Object,
+                new[] { log4NetInput },
+                new IFilter[0],
+                new[] { new EventSink(mockOutput.Object, new IFilter[0]) }))
+            {
+                var logger = LogManager.GetLogger("EventFlow", typeof(Log4NetInputTests));
+                logger.Info("some message");
+            }
+            mockOutput.Verify(
+                output => output.SendEventsAsync(It.Is<IReadOnlyCollection<EventData>>(c => c.Count == 1),
+                    It.IsAny<long>(), It.IsAny<CancellationToken>()), Times.Exactly(1));
+        }
+
+        [Fact]
+        public void LowerLevelDefinedDoesNotWriteToPipeline()
+        {
+            var healthReportMock = new Mock<IHealthReporter>();
+            var configurationMock = new Log4netConfiguration { Log4netLevel = Level.Info };
+            var mockOutput = new Mock<IOutput>();
+
+            using (var log4NetInput = new Log4netInput(configurationMock, healthReportMock.Object))
+            using (var pipeline = new DiagnosticPipeline(
+                healthReportMock.Object,
+                new[] { log4NetInput },
+                new IFilter[0],
+                new[] { new EventSink(mockOutput.Object, new IFilter[0]) }))
+            {
+                var logger = LogManager.GetLogger("EventFlow", typeof(Log4NetInputTests));
+                logger.Debug("some message");
+            }
+            mockOutput.Verify(
+                output => output.SendEventsAsync(It.Is<IReadOnlyCollection<EventData>>(c => c.Count == 1),
```

Comment:
Shouldn't the collection of events passed to `SendEventsAsync` be just `It.IsAny<IReadOnlyCollection<EventData>>()`?

After all we do not want any events being sent from this test, period.

Reply:
I agree, it should be as you suggested. Great catch. =)

IsAny is a much better option.